cibyl/orchestrator.py

```diff
@@ -74,11 +74,21 @@ def create_ci_environments(self) -> None:
             raise InvalidConfiguration from exception
 
     @staticmethod
-    def populate(environment, model_instances):
-        """Populate environment instance with the provided model instances"""
-        for model_instance in model_instances:
-            LOG.debug("populating environment %s: %s",
-                      environment, model_instance)
+    def populate(system, model_instances):
+        """Populate system instance with the provided model instances.
+
+        :param system: System to add the models to
+        :type system: :class:`.System`
+        :param model_instances: List of models to include in the system
+        :type model_instances: list
+        """
+        LOG.debug("populating system %s", system)
+        for attribute_key, value in model_instances.items():
+            population_method_name = f"add_{attribute_key}"
```

Comment:
Wouldn't it be simpler here to add a populate method to System classes that calls the appropiate add_* method? This way we also wouldn't need to specify the type of model in the return value of the source method called

Reply:
Interesting. So something like that?

System.add()   -> System.add_jobs() ?
But how System.add know to call System.add_jobs if we don't pass the type of the model? Can you write the signature of the methods?

Reply:
` def add(self, model_instances):                                                                                                                                                     
         for model in model_instances:                                                                                                                                                   
             if not isinstance(model, Job):                                                                                                                                              
                 continue                                                                                                                                                                
             self.add_jobs(model) `

Something like this should work, although it has the drawback of checking the type of each model, maybe we can keep the return format from the source and use the key to check, but hiding it from the orchestrator.

Reply:
Yeah, checking the type of 1000+ objects is overhead we can avoid.